### PR TITLE
libfabric: Add CUDA variant

### PR DESCRIPTION
**Pull request by jack-morrison**
Adds `CudaPackage` to libfabric for supporting the `--with-cuda` configure argument.

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
LGTM.  Would like to hear from @rajachan as to whether the changes are sufficient.

**Review by rajachan — APPROVED** [review]

File: var/spack/repos/builtin/packages/libfabric/package.py [hunk 1]
```diff
@@ -211,6 +211,9 @@ def configure_args(self):
             else:
                 args.append("--enable-{0}=no".format(fabric))
 
+        if self.spec.satisfies("+cuda"):
```
Comment (rajachan) [inline var/spack/repos/builtin/packages/libfabric/package.py:214]: Might as well add the logic for other HMEM interfaces in this manner?

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/libfabric/package.py:214]: Ping me in a comment when this PR is ready to merge then.

Comment (jack-morrison) [reply var/spack/repos/builtin/packages/libfabric/package.py:214]: Thanks @rajachan!

I am happy to do the same for ROCm, but would it be acceptable to do so as a separate PR? If that's ok, I will try to get to it soon.

Comment (rajachan) [reply var/spack/repos/builtin/packages/libfabric/package.py:214]: Sounds good. @tldahlgren this is ready to go then.

**Pull request merged**